lib/std/fs/Dir.zig

```diff
@@ -2683,9 +2747,11 @@ pub const StatFileError = File.OpenError || File.StatError || posix.FStatAtError
 /// On other platforms, `sub_path` is an opaque sequence of bytes with no particular encoding.
 pub fn statFile(self: Dir, sub_path: []const u8) StatFileError!Stat {
     if (native_os == .windows) {
-        var file = try self.openFile(sub_path, .{});
+        // note: it is possible to stat the file directly using `NtQueryInformatioByName`
+        // see PR #20843 for why it has not been implemented
```

Comment:
This comment should either include the explanation, or the comment should be deleted.

Related: I can't seem to get `NtQueryInformationByName` to actually work when using `OBJ_OPENLINK`, it always returns `INVALID_PARAMETER` for me. Also had to switch to `FILE_STAT_INFORMATION` to avoid `INVALID_PARAMETER`

This code works when `follow_symlinks` is true, but hits `INVALID_PARAMETER` when `follow_symlinks` is false:

<details>
<summary>Test code</summary>

```zig
const std = @import("std");
const windows = std.os.windows;

pub extern "ntdll" fn NtQueryInformationByName(
    ObjectAttributes: *windows.OBJECT_ATTRIBUTES,
    IoStatusBlock: *windows.IO_STATUS_BLOCK,
    FileInformation: *anyopaque,
    Length: windows.ULONG,
    FileInformationClass: windows.FILE_INFORMATION_CLASS,
) callconv(windows.WINAPI) windows.NTSTATUS;

const FILE_STAT_INFORMATION = extern struct {
    FileId: windows.LARGE_INTEGER,
    CreationTime: windows.LARGE_INTEGER,
    LastAccessTime: windows.LARGE_INTEGER,
    LastWriteTime: windows.LARGE_INTEGER,
    ChangeTime: windows.LARGE_INTEGER,
    AllocationSize: windows.LARGE_INTEGER,
    EndOfFile: windows.LARGE_INTEGER,
    FileAttributes: windows.ULONG,
    ReparseTag: windows.ULONG,
    NumberOfLinks: windows.ULONG,
    EffectiveAccess: windows.ACCESS_MASK,
};

pub fn main() !void {
    const dir = std.fs.cwd();
    const path = "testfile";
    {
        var file = try dir.createFile(path, .{});
        file.close();
    }
    const sub_path_w = std.unicode.utf8ToUtf16LeStringLiteral(path);
    const path_space = try windows.wToPrefixedFileW(dir.fd, sub_path_w);
    const path_len_bytes: u16 = @intCast(path_space.span().len * 2);
    const follow_symlinks = false;

    var nt_name = windows.UNICODE_STRING{
        .Length = path_len_bytes,
        .MaximumLength = path_len_bytes,
        .Buffer = @constCast(path_space.span().ptr),
    };
    // windows.OBJ_OPENLINK means it opens the link directly rather than its target
    const attributes: u32 = if (follow_symlinks) 0 else windows.OBJ_OPENLINK;
    var attr = windows.OBJECT_ATTRIBUTES{
        .Length = @sizeOf(windows.OBJECT_ATTRIBUTES),
        .RootDirectory = dir.fd,
        .Attributes = attributes,
        .ObjectName = &nt_name,
        .SecurityDescriptor = null,
        .SecurityQualityOfService = null,
    };
    var io_status_block: windows.IO_STATUS_BLOCK = undefined;
    var info: FILE_STAT_INFORMATION = undefined;

    const rc = NtQueryInformationByName(
        &attr,
        &io_status_block,
        &info,
        @sizeOf(FILE_STAT_INFORMATION),
        .FileStatInformation,
    );
    switch (rc) {
        .SUCCESS => {},
        .BUFFER_OVERFLOW => {},
        .INVALID_PARAMETER => unreachable,
        else => return windows.unexpectedStatus(rc),
    }

    std.debug.print("{any}\n", .{info});
}
```

</details>

If `OBJ_OPENLINK` can't actually be used with `NtQueryInformationByName`, then this comment should mention that.

Reply:
I currently do not have a windows device to test it, unfortunatly, so I will remove the comment. I will also remove the import of `NtQueryInformationByName` since it is not available for windows < Windows 10, version 1703, and i am not sure which windows versions are supported by zig.